Spanner/src/Transaction.php

```diff
@@ -538,9 +555,25 @@ public function executeUpdateBatch(array $statements, array $options = [])
         }
         $options['seqno'] = $this->seqno;
         $this->seqno++;
+    
+        $options['transactionType'] = $this->context;
+        if (is_array($this->transactionId) and isset($this->transactionId['begin'])) {
+            $options['begin'] = $this->transactionId['begin'];
+        } else {
+            $options['transactionId'] = $this->transactionId;
+        }
+        $selector = $this->transactionSelector($options);
+
+        $options['transaction'] = $selector[0];
```

Comment:
This block of code exists twice in `Transaction`, so it should be split into a separate function so as not to repeat code (including the previous block above it). Something like this would work:

```php
public function buildUpdateOptions(array $options): array
{
    unset($options['requestOptions']['transactionTag']);
    if (isset($this->tag)) {
        $options += [
            'requestOptions' => []
        ];
        $options['requestOptions']['transactionTag'] = $this->tag;
    }
    $options['seqno'] = $this->seqno;
    $this->seqno++;

    $options['transactionType'] = $this->context;
    if (is_array($this->transactionId) and isset($this->transactionId['begin'])) {
        $options['begin'] = $this->transactionId['begin'];
    } else {
        $options['transactionId'] = $this->transactionId;
    }
    $selector = $this->transactionSelector($options);

    $options['transaction'] = $selector[0];
    return $options;
}
```

... Although we will want to clean some things up in how this is done (e.g. `$this->transactionId` being multiple types), at least this way it's DRY.

Reply:
Except this comment, I have incorporated all others.